Modules/posixmodule.c

```diff
@@ -5088,212 +5088,243 @@ os__path_splitroot_impl(PyObject *module, path_t *path)
 }
 
 
+#define PY_IFREG 1
+#define PY_IFDIR 2
+#define PY_IFLNK 3
+#define PY_IFMNT 4
+
+static BOOL
+_testFileTypeByHandle(HANDLE hfile, int refFileType, BOOL diskOnly)
+{
+    assert(refFileType == PY_IFREG || refFileType == PY_IFDIR ||
+           refFileType == PY_IFLNK || refFileType == PY_IFMNT);
+
+    DWORD fileDevType = GetFileType(hfile);
+    if ((fileDevType == FILE_TYPE_CHAR) ||
+        (fileDevType != FILE_TYPE_DISK && diskOnly) ||
+        (fileDevType == FILE_TYPE_UNKNOWN && GetLastError() != 0))
+    {
+        return FALSE;
+    }
+
+    FILE_INFO_BY_HANDLE_CLASS FileInformationClass;
+    if (refFileType == PY_IFLNK || refFileType == PY_IFMNT) {
+        FileInformationClass = FileAttributeTagInfo;
+        FILE_ATTRIBUTE_TAG_INFO info;
```

Comment:
This won't work since `info` is in a new scope, but you could declare `atInfo` and `baInfo` in the outer scope and set a `pInfo` void pointer depending on the info class.

Reply:
Let's just extract the attributes like you did previously.